lib/graphql/schema/finder.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+module GraphQL
+  class Schema
+    # Find schema members using string paths
+    #
+    # @example Finding object types
+    # MySchema.find("SomeObjectType")
+    #
+    # @example Finding fields
+    # MySchema.find("SomeObjectType.myField")
+    #
+    # @example Finding arguments
+    # MySchema.find("SomeObjectType.myField.anArgument")
+    #
+    # @example Finding directives
+    # MySchema.find("@include")
+    #
+    class Finder
+      class MemberNotFoundError < ArgumentError; end
+
+      def initialize(schema)
+        @schema = schema
+      end
+
+      def find(path)
```

Comment:
Perhaps add a docblock like

```ruby
# @param path [String] dot-separated path to find
# @raise [MemberNotFoundError] if path could not be found
```

My description may be missing the distinction between type/directive, but you know best how to describe that.